samples/openapi3/client/petstore/python-experimental/petstore_api/model_utils.py

```diff
@@ -148,7 +148,7 @@ def __new__(cls, *args, **kwargs):
         oneof_anyof_classes = []
         oneof_anyof_classes.extend(cls._composed_schemas.get('oneOf', ()))
         oneof_anyof_classes.extend(cls._composed_schemas.get('anyOf', ()))
-        new_cls = cls.get_discriminator_class(kwargs)
+        new_cls = get_discriminator_class(cls, kwargs, set())
```

Comment:
We may be passing in _visited_composed_classes in kwargs here, so we need to include it in the inputs to get_discriminator_class.
Can you extract _visited_composed_classes from kwargs and pass it in to get_discriminator_class here?